backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import CollectionMetadata, CollectionVersion, DatasetArtifact, DatasetMetadata, DatasetStatus, DatasetVersion
+
+
+class DatabaseProviderInterface:
+        
+    def initialize_canonical_collection(self, owner: str, collection_metadata: CollectionMetadata) -> CollectionVersion:
+        """
+        Initializes a canonical collection, generating a canonical collection_id and a new version_id.
+        Returns the newly created CollectionVersion
+        """
+        pass
+
+
+    def get_collection_version(self, version_id: str) -> CollectionVersion:
+        """
+        Retrieves a specific collection version by id
+        """
+        pass
+
+    def get_all_versions_for_collection(self, collection_id: str) -> List[CollectionVersion]:
+        """
+        Retrieves all versions for a specific collections, without filtering
+        """
+        pass
+
+    def get_all_collections(self) -> Iterable[CollectionVersion]:
+        """
+        Retrieves all collections.
+        # TODO: add a filtering object.
+        """
+        pass
+
+    def delete_canonical_collection(self, collection_id: str) -> None:
```

Comment:
Perhaps tombstoning should remain a business layer concern. It seems asymmetric for the persistence layer to support a _logical_ deletion, where the data still exists physically, but then cannot later be found via the persistence layer. 

(It's also not clear to me yet how the physical model will support logical deletion. Presumably the "collection mapping" table would have a tombstoned flag?)